tests/unit/llvm_lit_tests/CycleEquivalence.ll

```diff
@@ -0,0 +1,187 @@
+;
+; Copyright rev.ng Labs Srl. See LICENSE.md for details.
+;
+
+; RUN: %revngopt %s -cycle-equivalence -debug-log=cycle-equivalence -o /dev/null |& FileCheck %s
+
+; while test
+
+define void @f(i32 noundef %a) {
+entry:
+  br label %while.cond
+
+while.cond:                                       ; preds = %while.body, %entry
+  %cmp = icmp slt i32 %a, 10
+  br i1 %cmp, label %while.body, label %while.end
+
+while.body:                                       ; preds = %while.cond
+  br label %while.cond
+
+while.end:                                        ; preds = %while.cond
+  ret void
+}
+
+; CHECK: [cycle-equivalence] Ordered Bracket Analysis Results:
+; CHECK: [cycle-equivalence] Bracket Equivalence Class ID: 0
+; CHECK: [cycle-equivalence] while.cond -> while.body
+; CHECK: [cycle-equivalence] Bracket Equivalence Class ID: 1
+; CHECK: [cycle-equivalence] while.cond -> while.end
+; CHECK: [cycle-equivalence] entry -> while.cond
+; CHECK: [cycle-equivalence]
```

Comment:
Right, I tried to minimize the test removing, for example, the blank lines and the name of the logger. I also tried to make the checks more robust with the use of  `CHECK-LABEL` and `CHECK-NEXT` in place of the simple `CHECK`, in order to avoid order-related false negatives. This may trigger order-related false positives, but since we are also interested in determinism, I think that we should take a look at possible order-related false positives should they arise